src/py21cmfast/inputs.py

```diff
@@ -659,6 +663,17 @@ def INHOMO_RECO(self):
         else:
             return self._INHOMO_RECO
 
+    @property
+    def FAST_FCOLL_TABLES(self):
+        """Automatically setting FAST_FCOLL_TABLES to True if USE_MINI_HALOS. For speed."""
+        if self.USE_MINI_HALOS and not self._FAST_FCOLL_TABLES:
+            logger.warning(
+                "You have set USE_MINI_HALOS to True but FAST_FCOLL_TABLES to False, so the code will be very slow. Setting to true"
+            )
```

Comment:
I don't think this should automatically be set if USE_MINI_HALOS is set. It just caused my run to crash when I set FAST_FCOLL_TABLES=False. Which should be a perfectly legitimate option.

I think a default should be selected (I think FAST_FCOLL_TABLES = FALSE) should be the default, with the option of speeding it up with setting this to true. Setting it to always return True when USE_MIN

Reply:
Great comments, thanks! I have moved USE_INTERPOLATION_TABLES to UserParams, and left the default as FALSE. If TRUE it checks that USE_INTERPOLATION_TABLES is TRUE too.

Reply:
Question, are the initial conditions re-ran automatically if anything is changed in UserParams? Or only for DIM and HII_DIM, etc? If it's the former we should not have FAST_FCOLL_TABLES (or USE_INTERPOLATION_TABLES) in there

Reply:
@JulianBMunoz they are re-run if any of the parameters in `UserParams` are modified, so we're all good.

Reply:
But this is a problem, isn't it? We don't necessarily want to re-run the ICs if FAST_FCOLL_TABLES or USE_INTERPOLATION_TABLES change, right?

Also, I just re-ran the same case with all the new changes and it still takes ~900 s for a tiny box (so its almost entirely table cost), so I did not see any improvement in speed.

Reply:
Ah, I see what you mean. Yeah, this is a more fundamental problem with the setup, and why I have previously advocated for having an `options` and `params` structure for each major computational component, so that the dependencies can be tracked with more granularity. I think there are ways of tuning this at the moment, but it's also probably not a massive problem.

Reply:
Ok this seems like a separate issue we can address at a different point if we so wish. For now it makes sense that FAST_FCOLL_TABLES sticks in the same category as USE_INTERPOLATION_TABLES. Thanks y'all!